src/test/java/org/broadinstitute/hellbender/engine/MultiVariantDataSourceUnitTest.java

```diff
@@ -176,16 +176,20 @@ public void testSerialQueries() {
 
             Iterator<VariantContext> it = multiVariantSource.query(new SimpleInterval("1", 1, 1200));
             while (it.hasNext()) {
-                it.next();
+                VariantContext vc = it.next();
                 count++;
+
+                Assert.assertTrue(vc.getSource().contains("interleavedVariants_"));
             };
             Assert.assertEquals(count, 14);
 
             count = 0;
             it = multiVariantSource.query(new SimpleInterval("2", 200, 600));
             while (it.hasNext()) {
-                it.next();
+                VariantContext vc = it.next();
                 count++;
+
+                Assert.assertTrue(vc.getSource().contains("interleavedVariants_"));
```

Comment:
This needs a test that verifies that the source names for the variant are propagated correctly. It looks like the test files used in this test method have no common start positions, so one way to do it would be to (in a separate test) first read the two vcfs in and populate a map from start position to source name, and then use MultiVariantDataSource to read them and validate each variant's source against the map by using it's position .

Reply:
ok

Reply:
ok